pallets/sto/src/lib.rs

```diff
@@ -378,8 +377,9 @@ decl_module! {
             }
 
             ensure!(remaining == 0.into(), Error::<T>::InsufficientTokensRemaining);
+            ensure!(cost >= fundraiser.minimum_investment, Error::<T>::InvestmentAmountTooLow);
             ensure!(
-                max_price.map(|max_price| cost <= max_price * investment_amount).unwrap_or(true),
+                max_price.map(|max_price| cost <= max_price.saturating_mul(purchase_amount) / price_divisor).unwrap_or(true),
```

Comment:
Should we check that max_price isn't Some(0) to avoid div by zero?

Reply:
`max_price` is in the numerator, not the denominator so I don't think this is an issue.

Reply:
Not related with this changes but It looks like `price_divisor` should be a common constant.